pkg/k8s/watchers/node.go

```diff
@@ -58,8 +54,8 @@ func (k *K8sWatcher) NodesInit(k8sClient *k8s.K8sClient) {
 						if hasAgentNotReadyTaint(node) || !k8s.HasCiliumIsUpCondition(node) {
 							k8sClient.ReMarkNodeReady()
 						}
-						err := k.updateK8sNodeV1(nil, node)
-						k.K8sEventProcessed(metricNode, metricCreate, err == nil)
+						errs := k.NodeChain.OnAddNode(node)
+						k.K8sEventProcessed(metricNode, metricCreate, errs == nil)
```

Comment:
I'm realizing that in the `OnAddNode` func, we initialize the `errors` slice that's returned here, so it will never be `nil`. We should change the call to `k.K8sEventProcessed(metricNode, metricCreate, len(errs) == 0)`

Reply:
Errors is no longer a slice, its collpased into a single error string that reports multiple errors, to remain in go's idiom for error handling.